int-set/src/lib.rs

```diff
@@ -498,6 +552,79 @@ where
     }
 }
 
+struct IterAfter<SetIter, AllValuesIter>
```

Comment:
naively I'm surprised we need to define a new type here? I would think that 'iter after' could be implemented with the same type as normal iteration, and it just being a matter of setting up the correct starting state.

Reply:
The problem is the existing Iter type works specifically with DoubleEndedIterator's. For simplicity I've made iter_after() not double ended, but unfortunately that means with need a new iter impl to deal with it.

Reply:
Actually thanks to your comment below about not bounding the types on the struct I think I can make this work reusing Iter